### ISSUE TITLE
failing on dpdk pkg-config check if no pkg-config is available

### ISSUE THREAD

**Issue by cpaelzer** [collectd/collectd#2404]
*   Version of collectd: head
*   Operating system / distribution: Ubuntu/Debian

## Expected behavior

Build without pkg-config

## Actual behavior

Fails since PR 2400

## Steps to reproduce

* build

## Background and solution

Due to a recent change (my fault in [PR 2400](https://github.com/collectd/collectd/pull/2400)) the Ci is now failing all the time.
See for example this [CI log](https://ci.collectd.org/job/pull-requests-prepare-tarball/1844/console)

```
checking for DPDK... no
configure: error: Package requirements (libdpdk) were not met:

No package 'libdpdk' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables DPDK_CFLAGS
and DPDK_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
Build step 'Execute shell' marked build as failure
Archiving artifacts
Finished: FAILURE

```

The reason is that while working correctly in the pkg-config case no matter if old or new dpdk headers are used.
It has issues if no pkg-config is provided at all.
The reason for that is that the macro for the check errors out if there is no definition for the "else" part.
The fix is easy and a PR will be here soon.

**Comment by cpaelzer** [comment]
The linked PR fixes the case for me, please accept my excuse for breaking it in the first place and take a look.

**Comment by rubenk** [comment]
Fixed by 9e6bf7d
